### ISSUE TITLE
OSCAP scan false positives

### ISSUE THREAD

**Issue by polachz** [konstruktoid/ansible-role-hardening#42]
Hallo.

I have checked hardened VM instance of the RHLE/CentOS8 by [OSCAP](https://www.open-scap.org/tools/scap-workbench/) security scanner with the **CIS Red Hat Enterprise Linux 8 Benchmark** policy.

I have got many notices related to the auditd config. They are caused in many cases only by improper line check (fore example it expects **auid=unset** instead of the **auid=-1**)

I have changed the **hardening.rules.j2** to remove these problems. But I can ask you if you use some simmilar checks. 

Maybe my fix will repair Centos/Rhel 8 but can brings Centos 7 to troubles. 

So can be better to make decision if you can include these changes to the role and if yes, thenif you can do that globally, or can you provide different file versions for different OS families and versions.

Looking forward to your answer

Best Regards, Zdenek

**Comment by konstruktoid** [comment]
Hi again!
Yeah, I usually use the OSCAP scanners on RHEL/CentOS as well, and the CIS-CAT Assessor (https://www.cisecurity.org/cybersecurity-tools/cis-cat-pro/) on Ubuntu, Debian and RHEL/CentOS. The issue is that not one of these tools or related profiles really works in a mixed environment.

OSCAP lacks proper profile support regarding Ubuntu and Debian, see https://github.com/ComplianceAsCode/content/issues/6112, and the CIS scanner supports all those distributions but the various profiles have different ways of verifying the audit rules and I've opened multiple issues regarding this (e.g. https://workbench.cisecurity.org/community/14/tickets/11357).

For example when running the CIS scanner, the test `-k actions` will pass on Ubuntu since `^-w\s+/etc/sudoers.d(/)*\s+-p\s+wa\s+-k\s+\S+ *$` is checked.
The same rule will fail 4.1.3 on Centos and RedHat since `^-w /etc/sudoers -p wa -k scope$`, with the `-k scope`, is used as regex.

And regarding `auid`, you can use either "unset" which is easy to remember, or -1, or 4294967295. They are all equivalent.

In short, `auditd` rules and compliance in a mixed environment is a mess.

**Comment by polachz** [comment]
OK. Then best solution here is to make apply the role _hardening.rules.j2_ optional with default state true, I think.

It allows to use suitable rulesset for person who do not care about OSCAP or other checks and for people who use a scanner it allows to apply whatever they need in compliant form...

Do you agree ?

**Comment by konstruktoid** [comment]
> OK. Then best solution here is to make apply the role _hardening.rules.j2_ optional with default state true, I think.

What do you mean?

**Comment by polachz** [comment]
My suggestion is to add a variable define_auditd_rules and set it true as default.

If the variable is true, the hardening.rules.j2 template will be used
If the variable is false, the hardening.rules.j2 template will be skipped

It allows to use another task to define auditd rules if user want but preserve role behavior in current state as default

**Comment by konstruktoid** [comment]
Yeah, seems reasonable.

**Comment by polachz** [comment]
Ok, I'll prepare pull request tomorrow :)
